cmd/scale.go

```diff
@@ -394,6 +394,12 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 		templateJSON["variables"].(map[string]interface{})[sc.agentPool.Name+"VMNamePrefix"] = sc.containerService.Properties.GetAgentVMPrefix(sc.agentPool, winPoolIndex)
 	}
 	if orchestratorInfo.OrchestratorType == api.Kubernetes {
+		if orchestratorInfo.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku {
+			err = transformer.NormalizeForK8sSLBScalingOrUpgrade(sc.logger, templateJSON)
```

Comment:
since AKS won't call this path, how will AKS scale work with this change?

Reply:
@yangl900 @xuto2 PTAL

Reply:
AKS scale works because 
1. AKS always pass AKS generated LBBackendAddressPoolIDs to AKS-E when scaling/upgrading a SLB backed agent nodes,
2. The below change in pkg/engine/armresources.go in this PR (pasted below):

if !cs.Properties.OrchestratorProfile.IsPrivateCluster() && 
		!cs.Properties.AnyAgentHasLoadBalancerBackendAddressPoolIDs() && 
		cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku { 
		publicIPAddress := CreatePublicIPAddress(false) 
		loadBalancer := CreateAgentLoadBalancer(cs.Properties, true)